interfaces/cython/cantera/composite.py

```diff
@@ -666,14 +690,30 @@ def append(self, state=None, **kwargs):
                 ) from None
             setattr(self._phase, attr, [kwargs[a] for a in attr])
 
-        self._states.append(self._phase.state)
-        self._indices.append(len(self._indices))
+        extra = self._extra
```

Comment:
This doesn't actually create a copy of `self._extra`, so using this to revert changes on erroneous input doesn't work. For example:
```python
>>> gas = ct.Solution('air.yaml')
>>> states = ct.SolutionArray(gas, 3, extra={"p1": [5, 6, 7], "p2": [1, 2, 3]})
>>> states.append(TPX=(300, 101325, 'O2:1.0'), p1=5, p2=[7,8])
ValueError: Encountered incompatible value '[7, 8]' for extra column 'p2'.
>>> print(states._extra)
OrderedDict([('p1', array([5, 6, 7, 5])), ('p2', array([1, 2, 3]))])
```

Reply:
Good point - it's all by reference in Python, of course.